### DISCUSSION TITLE
Arrays for segment "text" templates

### DISCUSSION THREAD

**Discussion by RustyBrakes** [JanDeDobbeleer/oh-my-posh#1615] [Ideas]
I'd ideally like the "color_template" format with arrays to work for the content "text templates" too. I think the best way to explain this one is with an example. Lets say I want to use different icons based on template properties of the battery segment. Currently I would do:

`"template": "{{if lt 10 .Percentage}}\uf579{{else}}{{if lt 30 .Percentage}}\uf57b{{else}}{{if lt 50 .Percentage}}\uf57d{{else}}{{if lt 70 .Percentage}}\uf57f{{else}}{{if lt 90 .Percentage}}\uf581{{else}}\uf583{{end}}{{end}}{{end}}{{end}}{{end}}"`

And what I'd like to do is:

```
"template": [
  "{{if lt 10 .Percentage}}\uf579{{end}}",
  "{{if lt 30 .Percentage}}\uf57b{{end}}",
  "{{if lt 50 .Percentage}}\uf57d{{end}}",
  "{{if lt 70 .Percentage}}\uf57f{{end}}",
  "{{if lt 90 .Percentage}}\uf581{{end}}",
  "\uf583"
]
```

It only really provides an advantage when the template text will be X or Y or Z, but I have already found a few use cases when that would help and make the code much more readable.

I'm very open to feedback if this is a bad idea!

**Comment by JanDeDobbeleer** [comment]
I agree. BUT, can I please first refactor everything to have a template? Then I can move that logic to a single spot rather than everywhere.

**Reply by RustyBrakes** [reply]
Jan, of course you can! I really have no rush. I wish I had any sort of training and experience to be able to help you, and I'm sorry that when I make suggestions they sound like demands. They are not meant like that, just some things that would make a brilliant project even better

**Reply by JanDeDobbeleer** [reply]
It's OK, I treat almost every request as a demand/task if it fits a proper use-case.
